pandas/core/groupby/groupby.py

```diff
@@ -3219,6 +3226,55 @@ def max(
                 npfunc=np.max,
             )
 
+    def _idxmax_idxmin(
+        self,
+        how: Literal["idxmin", "idxmax"],
+        alt: Callable,
+        numeric_only: bool = False,
+    ) -> Series | DataFrame:
+        """Compute idxmax/idxmin.
+
+        Parameters
+        ----------
+        how: {"idxmin", "idxmax"}
+            Whether to compute idxmin or idxmax.
+        alt: Callable
+            Fallback for non-Cython (e.g. object) types.
+        numeric_only : bool, default False
+            Include only float, int, boolean columns.
+
+        Returns
+        -------
+        Series or DataFrame
+            idxmax or idxmin for the groupby operation.
+        """
+
+        def post_process(x):
+            if x.size == 0:
+                result = x.astype(self.obj.index.dtype)
```

Comment:
x.dtype should be np.intp right?  could this fail if self.obj.index is EA?

Reply:
> x.dtype should be np.intp right?

Yes

> could this fail if self.obj.index is EA?

I don't think so - it's tested in `test_empty_groupby` with Int64, Float64, and boolean.